src/gbm/gbtree_model.h

```diff
@@ -37,12 +45,16 @@ struct GBTreeModelParam : public dmlc::Parameter<GBTreeModelParam> {
   int reserved[32];
   /*! \brief constructor */
   GBTreeModelParam() {
-    std::memset(this, 0, sizeof(GBTreeModelParam));
+    std::memset(this, 0, sizeof(GBTreeModelParam));  // FIXME(trivialfis): Why?
```

Comment:
C++ standard does not require a compiler to generate initialization of integers (does not set them to zero).
So they will have a value whatever was in memory before allocating

Reply:
@trams I'm aware of this.  Just don't like mixing c style and c++ style of doing things.  Plus it's not idea to have multiple initialization methods used in one place.  So I should remove it when starting cleaning up.